### DISCUSSION TITLE
Set engines to all LTS supported versions of node and npm

### DISCUSSION THREAD

**Discussion by skjnldsv** [nextcloud/standards#7] [General]
From @korelstar 

> Devs can still use npm 8 if they want :)

But this will give warnings, since npm 8 is not compatible to `^7.0.0`. What about setting `NPM_VERSION: "^7.0.0 || ^8.0.0"`, then all node@16 installations (e.g. official docker container) will work just out of the box?

**Comment by skjnldsv** [comment]
My issue with that is maintenance.
We will need to monitor all LTS versions of node and npm and create 50 PRs to update the engines every time.
This creates an unnecessary burden for a minimal gain imho.

I'm personally ok keeping the warning. I know far too well what those kind of maintenance and monitoring generate as workload and I'd like to avoid it. LTS are maintained for 3 years, which means every 3 year we have to update all engines. Seems quite acceptable compared to every year?

**Comment by jotoeri** [comment]
( Resumption on https://github.com/nextcloud/standards/discussions/6#discussioncomment-3084562 )

Just as we keep the possibility of npm7, but the default 'out of the box' is mostly npm8, i personally think, that we should include `npm^8` as officially supported without a warning showing up.

So keeping node fixed to 16, and adding npm 7 & 8 for now, i don't think there is the necessity to monitor everything @skjnldsv. Node 16 with Npm 7 & 8 works for now, so the only thing that will need a change is if node 16 starts supporting npm 9. And i don't think we need to shoot faster than our shadow: Once this happens, any dev will come up with and we can just add `^9` to the npm versions then. Thus you don't have to monitor it on your own, but once it becomes important 'automatically' anybody will call up for it. 😉

**Reply by skjnldsv** [reply]
> Thus you don't have to monitor it on your own, but once it becomes important 'automatically' anybody will call up for it.

Well, the difference between the reality and the theory here.
From my experience, over the last years, it will be ignored by most and fall back onto my lap :shrug:

**Reply by jotoeri** [reply]
But don't we also train everybody to ignore warnings, if we keep now the npm8 warning intentionally? Feels a bit like a spiral, that we should try to exit, to get you out of that liability? 🤔
